trl/models/__init__.py

```diff
@@ -13,8 +13,24 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from ..import_utils import is_peft_available
 from .modeling_base import PreTrainedModelWrapper, create_reference_model
 from .modeling_value_head import AutoModelForCausalLMWithValueHead, AutoModelForSeq2SeqLMWithValueHead
 
 
-SUPPORTED_ARCHITECTURES = (AutoModelForCausalLMWithValueHead, AutoModelForSeq2SeqLMWithValueHead)
+if is_peft_available():
+    from peft.peft_model import PeftModelForCausalLM, PeftModelForSeq2SeqLM
+
+SUPPORTED_ARCHITECTURES = (
+    (
+        AutoModelForCausalLMWithValueHead,
+        AutoModelForSeq2SeqLMWithValueHead,
+    )
+    if not is_peft_available()
+    else (
+        AutoModelForCausalLMWithValueHead,
+        AutoModelForSeq2SeqLMWithValueHead,
+        PeftModelForCausalLM,
+        PeftModelForSeq2SeqLM,
```

Comment:
Wouldn't we need a value head for those models as well?

Reply:
No I don't think so, as we only use peft on the pretrained model.

Reply:
Fixed in [e2aff14](https://github.com/lvwerra/trl/pull/163/commits/e2aff14f2ccee3099d2eeb2af204d6b99b39ffb0)